sklearn/metrics/_ranking.py

```diff
@@ -221,6 +221,11 @@ def average_precision_score(
     def _binary_uninterpolated_average_precision(
         y_true, y_score, pos_label=1, sample_weight=None
     ):
+        if y_true.shape[0] < 2:
```

Comment:
Unit tests seem to fail because `y_true` can be a python list without a `.shape` argument (I think, I am not a scikit-learn maintainer)